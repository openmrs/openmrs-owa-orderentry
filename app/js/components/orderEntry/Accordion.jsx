import React from 'react';
import PropTypes from 'prop-types';

class Accordion extends React.Component {
  state = {
    isVisible: this.props.open === undefined ? false : this.props.open,
  };

  render() {
    return (
      <React.Fragment>
        <tr
          className="accordion accordion-link"
          role="button"
          tabIndex={0}
          onClick={() => {
            this.setState({ isVisible: !this.state.isVisible });
          }}>
          <th className="th-1 th-invisible">
            <span>
              {this.state.isVisible ? (
                <i className="toggle-icon icon-caret-down small rotate90" />
              ) : (
                <i className="toggle-icon icon-caret-right small rotate90" />
              )}
            </span>
            27 jun 2018
          </th>
          {this.props.title}
        </tr>
        <tr>
          <td className="th-invisible" />
          <td className="th-details">
            <div
              className={`th-details accordion-two content ${
                !this.state.isVisible ? 'close' : 'open'
              }`}>
              {this.props.children}
            </div>
          </td>
          <td className="th-invisible" />
          <td className="th-invisible" />
        </tr>
      </React.Fragment>
    );
  }
}

Accordion.propTypes = {
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  title: PropTypes.object.isRequired,
};

Accordion.defaultProps = {
  open: false,
};

export default Accordion;
