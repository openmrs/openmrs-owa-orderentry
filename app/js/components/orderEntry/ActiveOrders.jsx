import React from 'react';
import { connect } from 'react-redux';
import format from 'date-fns/format';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { activeOrderAction } from '../../actions/activeOrderAction';
import { addDraftOrder } from '../../actions/draftTableAction';
import { setOrderAction } from '../../actions/orderAction';
import imageLoader from '../../../img/loading.gif';


export class ActiveOrders extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      draftOrder: {},
      pageCount: 0,
      limit: 10,
      startIndex: 0,
      pageNumber: 0,
    };
    this.onPageChange = this.onPageChange.bind(this);
  }
  componentDidMount() {
    this.fetchActiveOrders(this.props);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.tabName !== this.props.tabName) {
      this.fetchActiveOrders(nextProps);
    }
    this.setState({ pageCount: nextProps.pageCount });
  }

  onClickDiscontinue = (order) => {
    const {
      uuid,
      drug,
      dose,
      doseUnits,
      frequency,
      route,
      duration,
      careSetting,
      durationUnits,
      asNeededCondition,
      orderer,
      dosingInstructions,
      dosingType,
      type,
      quantity,
      quantityUnits,
      orderNumber,
    } = order;

    this.setState({
      draftOrder: {
        uuid,
        careSetting: careSetting.uuid,
        drugName: drug.display,
        action: 'DISCONTINUE',
        dose,
        drug: drug.uuid,
        dosingUnit: (doseUnits && doseUnits.display) || doseUnits,
        frequency: (frequency && frequency.display) || frequency,
        route: (route && route.display) || route,
        duration,
        type,
        orderer: orderer.uuid,
        durationUnit: (durationUnits && durationUnits.display) || durationUnits,
        dosingType,
        reason: asNeededCondition,
        drugInstructions: dosingInstructions,
        dispensingQuantity: quantity,
        dispensingUnit: (quantityUnits && quantityUnits.display) || quantityUnits,
        orderNumber,
      },
    }, () => {
      this.props.onDelete(true);
      this.props.setOrderAction('DISCONTINUE', orderNumber);
      this.props.addDraftOrder(this.state.draftOrder);
    });
  }

  onPageChange({ selected }) {
    let { startIndex, pageNumber } = this.state;
    const { limit } = this.state;
    startIndex = Math.ceil(selected * limit);
    pageNumber = startIndex / limit;
    this.setState(() => ({ startIndex, pageNumber }));
    this.fetchActiveOrders(this.props, { limit, startIndex });
  }

  fetchActiveOrders(props, newState) {
    const { limit, startIndex = 0 } = newState || this.state;
    this.setState(() => ({ startIndex: 0 }));
    const { location, careSetting } = props;
    const query = new URLSearchParams(location.search);
    const patientUuid = query.get('patient');
    const caresettingUuid = careSetting.uuid;
    this.props.activeOrderAction(limit, startIndex, patientUuid, caresettingUuid);
  }

  showOrders = activeOrders => activeOrders
    .map((order) => {
      const {
        uuid,
        action,
        careSetting,
        dateActivated,
        autoExpireDate,
        drug,
        dose,
        doseUnits,
        frequency,
        route,
        duration,
        durationUnits,
        dosingInstructions,
        quantity,
        quantityUnits,
        dosingType,
        orderer,
        orderNumber,
        status,
      } = order;

      let details;


      if (dosingType === 'org.openmrs.SimpleDosingInstructions') {
        details = (
          <td>

            {drug.display}:
            {` ${dose}`}
            {` ${doseUnits.display}`}
            {`, ${frequency.display}`}
            {`, ${route.display}`}
            {duration && `, for ${duration}`}
            {durationUnits && ` ${durationUnits.display} total`}
            {dosingInstructions && `, (${dosingInstructions})`}
            {(quantity && quantityUnits) && `, (Dispense: ${quantity} ${quantityUnits.display})`}

          </td>
        );
      } else {
        details = (
          <td>

            {drug.display}:
            {dosingInstructions && ` ${dosingInstructions}`}
            {(quantity && quantityUnits) && `, (Dispense: ${quantity} ${quantityUnits.display})`}

          </td>
        );
      }

      let showStatus;

      if (status === 'EDIT' || status === 'DRAFT') {
        showStatus = (
          <p> will REVISE </p>
        );
      } else if (this.props.isDelete && status === 'DISCONTINUE') {
        showStatus = (
          <p> will DISCONTINUE </p>
        );
      } else {
        showStatus = (
          <div>
            <a
              href="#"
              onClick={() => this.props.handleEditActiveDrugOrder(order)}
            > <i className="icon-edit" title="Edit" />
            </a>
            <a > <i
              className="icon-remove icon-color"
              title="Delete"
              id="delete"
              role="button"
              tabIndex="0"
              onKeyPress={this.handleKeyPress}
              onClick={() => this.onClickDiscontinue(order)} />
            </a>
          </div>
        );
      }

      return (
        <tr key={uuid} >
          <td>
            {format(dateActivated, 'MM-DD-YYYY HH:mm')} {autoExpireDate && (`- ${format(autoExpireDate, 'MM-DD-YYYY HH:mm')}`)}
          </td>
          {details}
          <td className="text-center action-padding" >
            {showStatus}
          </td>
        </tr>
      );
    })


  render() {
    const { activeOrders, loading } = this.props.drugOrder;

    if (!activeOrders || loading) {
      return (
        <div className="text-align-center">
          <img src={imageLoader} alt="loader" />
        </div>
      );
    } else if (activeOrders.length === 0) {
      return (
        <div>
          <p>No Active Orders</p>
        </div>
      );
    }
    return (
      <div>
        <table className="table bordered mw-958-px">
          <thead>
            <tr>
              <th className="w-145-px">Date</th>
              <th>Details</th>
              <th className="w-81-px">Actions</th>
            </tr>
          </thead>
          <tbody>
            {this.showOrders(activeOrders)}
          </tbody>
        </table>
        <div className="clear">
          <div className="float-left-padding">
            {this.props.showResultCount}
          </div>
          <div className="dataTables_paginate">
            <ReactPaginate
              pageCount={this.state.pageCount}
              pageRangeDisplayed={5}
              marginPagesDisplayed={3}
              previousLabel="Previous"
              nextLabel="Next"
              breakClassName="text-align-center"
              initialPage={0}
              containerClassName="react-paginate-container"
              pageLinkClassName="page-link"
              activeClassName="active-link"
              disabledClassName="active-link"
              nextLinkClassName="page-link"
              previousLinkClassName="page-link"
              onPageChange={this.onPageChange}
              forcePage={this.state.pageNumber}
              disableInitialCallback
            />
          </div>
        </div>
      </div>
    );
  }
}

ActiveOrders.defaultProps = {
  showResultCount: 'Showing 1 to 10 of 55 entries',
  pageCount: 0,

};


const mapStateToProps = ({ activeOrderReducer }) => ({
  drugOrder: activeOrderReducer,
  pageCount: activeOrderReducer.pageCount,
  showResultCount: activeOrderReducer.showResultCount,
});

const mapDispatchToProps = dispatch => ({
  activeOrderAction: (limit, startIndex, careSetting, patientUuid) =>
    dispatch(activeOrderAction(limit, startIndex, careSetting, patientUuid)),
  addDraftOrder: order =>
    dispatch(addDraftOrder(order)),
  setOrderAction: (action, orderNumber) =>
    dispatch(setOrderAction(action, orderNumber)),
});

ActiveOrders.propTypes = {
  handleEditActiveDrugOrder: PropTypes.func.isRequired,
  tabName: PropTypes.string.isRequired,
  activeOrderAction: PropTypes.func.isRequired,
  addDraftOrder: PropTypes.func.isRequired,
  setOrderAction: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  isDelete: PropTypes.bool.isRequired,
  drugOrder: PropTypes.shape({
    loading: PropTypes.bool,
    activeOrders: PropTypes.arrayOf(PropTypes.shape({})),
  }),
  activeOrders: PropTypes.shape({}).isRequired,
  showResultCount: PropTypes.string,
  pageCount: PropTypes.number,
};

ActiveOrders.defaultProps = {
  drugOrder: {
    loading: false,
    activeOrders: [],
  },
};

export default connect(mapStateToProps, mapDispatchToProps)(ActiveOrders);
