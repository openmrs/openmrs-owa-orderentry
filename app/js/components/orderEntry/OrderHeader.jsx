import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const OrderHeader = ({ status, orderable }) => (
  <Fragment>
    <th className="th-invisible">{orderable}</th>
    <th className="th-invisible">{status}</th>
    <th className="th-invisible">
      <i className="icon-edit" />
      <i className="icon-delete" />
    </th>
  </Fragment>
);

export default OrderHeader;

OrderHeader.defaultProps = {
  status: '',
  orderable: '',
};

OrderHeader.propTypes = {
  status: PropTypes.string,
  orderable: PropTypes.string,
};
