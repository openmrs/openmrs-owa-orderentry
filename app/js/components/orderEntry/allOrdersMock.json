{
  "orders": [
    {
      "orderable": "Amoxycillin syrup",
      "date": "24/07/2018",
      "type": "drugorder",
      "dosingInstructions": "25mg of Amoxycillin syrup for the next 5 days",
      "dispense": "45",
      "activeDates": "25/07/2018 - 25/07/2019",
      "orderer": "Efosa Okpugie",
      "status": "Active",
      "id": 1
    },
    {
      "date": "24/12/2018",
      "orderable": "Paracetamol",
      "type": "drugorder",
      "dosingInstructions": "25mg of Amoxycillin syrup for the next 5 days",
      "dispense": "45",
      "activeDates": "25/07/2018 - 25/07/2019",
      "orderer": "Mark Goodrich",
      "status": "Active",
      "id": 2
    },
    {
      "date": "24/09/2018",
      "type": "testorder",
      "orderable": "Hematocrit",
      "urgency": "ROUTINE",
      "orderer": "Mark Goodrich",
      "status": "Active",
      "id": 3
    }
  ]
}
