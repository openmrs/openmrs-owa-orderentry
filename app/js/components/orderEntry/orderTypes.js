export const DRUG_ORDER = { id: 1, text: 'Drug Orders' };
export const LAB_ORDER = { id: 2, text: 'Lab Orders' };
export const ORDER_FROM_SETS = { id: 3, text: 'Order From Sets' };
