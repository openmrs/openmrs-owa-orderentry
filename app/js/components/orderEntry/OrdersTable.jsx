import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import swal from 'sweetalert';
import format from 'date-fns/format';
import Accordion from './Accordion';
import OrderHeader from './OrderHeader';
import LabOrderDetails from './LabOrderDetails';
import DrugOrderDetails from './DrugOrderDetails';
import fetchOrders from '../../actions/fetchOrders';
import { DRUG_ORDER } from './orderTypes';
import { setSelectedOrder } from '../../actions/orderAction';
import { discontinueOrder } from '../../actions/createOrder';
import { getConceptShortName } from '../../utils/helpers';

/* eslint-disable max-len */
export class OrdersTable extends PureComponent {
  state = {
    discontinueReason: '',
  };
  componentDidMount() {
    this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.patient.uuid !== prevProps.patient.uuid) {
      this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
    }
  }

  getUUID = (items, itemName) => items.find(item => item.display === itemName)

  setDiscontinuedOrder = async (order) => {
    const { intl } = this.props;
    const {
      orderNumber,
    } = order;

    const discontinueQuestion = intl.formatMessage({ id: "app.orders.discontinue.question", defaultMessage: "Are you sure you want to discontinue this order ?" });
    const discontinueMsg = intl.formatMessage({ id: "app.orders.discontinue", defaultMessage: "DISCONTINUE" });
    const reasonToDiscontinue = intl.formatMessage({ id: "app.orders.discontinue.reason", defaultMessage: "Reason for discontinuing" });
    const yesMsg = intl.formatMessage({ id: "reactcomponents.yes", defaultMessage: "YES" });
    const noMsg = intl.formatMessage({ id: "reactcomponents.no", defaultMessage: "NO" });

    const discontinueDecision = await swal(discontinueQuestion, {
      buttons: {
        YES: yesMsg,
        NO: noMsg,
      },
    });
    if (discontinueDecision === "YES") {
      const reason = await swal({
        text: reasonToDiscontinue,
        content: "input",
        button: {
          text: discontinueMsg,
          closeModal: true,
        },
      });
      const discontinuedOrder = order.orderType.name.match(/drug/i)
        ? this.formatDrugOrderData(order, reason)
        : this.formatLabOrderData(order);
      this.discontinueOrder(discontinuedOrder, orderNumber);
    }
  };

  formatDrugOrderData = (order, reason) => {
    const {
      drugDispensingUnits,
      drugDosingUnits,
      drugRoutes,
      durationUnits,
      orderFrequencies,
    } = this.props.allConfigurations;
    const {
      careSetting,
      dose,
      dosingUnit,
      drugInstructions,
      dosingType,
      drug,
      duration,
      durationUnit,
      frequency,
      orderer,
      dispensingQuantity,
      dispensingUnit,
      route,
      type,
      previousOrder,
    } = order;
    return {
      action: 'DISCONTINUE',
      asNeeded: (reason && true) || false,
      asNeededCondition: reason,
      autoExpireDate: null,
      orderReasonNonCoded: this.state.discontinueReason || null,
      careSetting,
      commentToFulfiller: '',
      dose: dose || null,
      doseUnits: (
        dosingUnit && this.getUUID(drugDosingUnits, dosingUnit).uuid) || null,
      dosingInstructions: drugInstructions || null,
      dosingType,
      drug,
      duration: duration || null,
      durationUnits: (
        durationUnit && this.getUUID(durationUnits, durationUnit).uuid) || null,
      frequency: (
        frequency && this.getUUID(orderFrequencies, frequency).uuid) || null,
      numRefills: 0,
      orderer,
      previousOrder,
      quantity: dispensingQuantity || null,
      quantityUnits: (
        dispensingUnit && this.getUUID(drugDispensingUnits, dispensingUnit).uuid) || null,
      route: (
        route && this.getUUID(drugRoutes, route).uuid) || null,
      type,
    };
  };

  formatLabOrderData = order => ({
    action: 'DISCONTINUE',
    concept: order.concept,
    careSetting: order.careSetting,
    encounter: this.props.encounterType.uuid,
    orderer: this.props.session.currentProvider.uuid,
    patient: this.props.patient.uuid,
    type: 'testorder',
    urgency: order.urgency || 'ROUTINE',
    previousOrder: order.uuid,
  });

  handleActiveOrderEdit = (order) => {
    this.props.dispatch(setSelectedOrder({ order: { ...order, status: 'EDIT' }, currentOrderType: DRUG_ORDER, activity: 'EDIT' }));
  }

  discontinueOrder = async (order, orderNumber) => {
    const payload = {
      encounterProviders: [{
        encounterRole: this.props.encounterRole.uuid,
        provider: this.props.sessionReducer.currentProvider.uuid,
      }],
      encounterType: this.props.encounterType.uuid,
      location: {
        uuid: this.props.sessionReducer.sessionLocation.uuid,
      },
      orders: [order],
      patient: this.props.patient.uuid,
    };
    await this.props.dispatch(discontinueOrder(payload, orderNumber));
  }

  renderNoFilterResults = () => {
    const {
      filteredOrders,
      status: { fetched },
    } = this.props;
    if (fetched && filteredOrders.length === 0) {
      return (
        <tr>
          <td className="th-invisible" />
          <td className="no-results th-invisible">
            <h4 className="no-result-info">No Orders</h4>
          </td>
          <td className="th-invisible" />
          <td className="th-invisible" />
        </tr>
      );
    }
    return null;
  };

  render() {
    const {
      filteredOrders, dateFormat, sessionReducer, orderType,
    } = this.props;

    const ordersToDisplay = filteredOrders.filter((order) => {
      if (orderType) {
        if (orderType === 'drugorders') {
          return order.orderType.name.match(/drug/i);
        } else if (orderType === 'laborders') {
          return order.orderType.name.match(/test/i);
        }
      }
      return true;
    });

    return (
      <React.Fragment>
        {ordersToDisplay &&
          ordersToDisplay.map(order => (
            <Accordion
              title={
                <OrderHeader
                  status="Active"
                  orderable={order.drug ? order.drug.display : getConceptShortName(order.concept, sessionReducer.locale)}
                  order={order}
                  handleEdit={this.handleActiveOrderEdit}
                  handleDiscontinue={this.setDiscontinuedOrder}
                />
              }
              key={order.uuid}
              dateFormat={dateFormat}
              date={order.dateActivated}>
              {order.orderType.name.match(/drug/i) ? (
                <DrugOrderDetails
                  dosingInstructions={order.dosingInstructions}
                  dispense={order.dispense}
                  activeDates={format(order.dateActivated, dateFormat)}
                  orderer={order.orderer.display.split('-')[1]}
                />
              ) : (
                <LabOrderDetails
                  urgency={order.urgency}
                  orderer={order.orderer.display.split('-')[1]}
                  activeDates={format(order.dateActivated, dateFormat)}
                  fulfillerStatus={order.fulfillerStatus}
                />
              )}
            </Accordion>
          ))}
        {this.renderNoFilterResults()}
      </React.Fragment>
    );
  }
}

OrdersTable.defaultProps = {
  dateFormat: '',
  filteredOrders: [],
  session: {
    currentProvider: {
      person: {
        uuid: '',
      },
    },
  },
};

OrdersTable.propTypes = {
  dispatch: PropTypes.func.isRequired,
  patient: PropTypes.shape({
    uuid: PropTypes.string,
  }).isRequired,
  filteredOrders: PropTypes.array,
  status: PropTypes.objectOf(PropTypes.bool).isRequired,
  dateFormat: PropTypes.string,
  allConfigurations: PropTypes.shape({
    drugDispensingUnits: PropTypes.arrayOf(PropTypes.any),
    drugDosingUnits: PropTypes.arrayOf(PropTypes.any),
    drugRoutes: PropTypes.arrayOf(PropTypes.any),
    durationUnits: PropTypes.arrayOf(PropTypes.any),
    orderFrequencies: PropTypes.arrayOf(PropTypes.any),
  }).isRequired,
  sessionReducer: PropTypes.shape({
    currentProvider: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    sessionLocation: PropTypes.shape({}),
  }).isRequired,
  encounterType: PropTypes.shape({
    uuid: PropTypes.string,
  }).isRequired,
  encounterRole: PropTypes.shape({
    uuid: PropTypes.string,
  }).isRequired,
  session: PropTypes.shape({
    currentProvider: PropTypes.shape({
      person: PropTypes.shape({
        uuid: PropTypes.string,
      }),
      uuid: PropTypes.string,
    }),
    currentLocation: PropTypes.object,
  }),
};

const mapStateToProps = state => ({
  allConfigurations: ((state.orderEntryConfigurations || {}).configurations || {}),
  careSetting: state.careSettingReducer.outpatientCareSetting,
  dateFormat: state.dateFormatReducer.dateFormat,
  encounterRole: state.encounterRoleReducer.encounterRole,
  encounterType: state.encounterReducer.encounterType,
  filteredOrders: state.fetchOrdersReducer.filteredOrders,
  inpatientCareSetting: state.careSettingReducer.inpatientCareSetting,
  orderType: state.contextReducer.orderType,
  outpatientCareSetting: state.careSettingReducer.outpatientCareSetting,
  patient: state.patientReducer.patient,
  session: state.openmrs.session,
  sessionReducer: state.openmrs.session,
  status: state.fetchOrdersReducer.status,
});

export default connect(mapStateToProps)(injectIntl(OrdersTable));
