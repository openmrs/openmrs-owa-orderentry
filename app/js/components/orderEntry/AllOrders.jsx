import React, { Component } from 'react';
import OrdersTable from './OrdersTable';

const AllOrders = () => (
  <div className="all-orders">
    <div className="orders-breadcrumb">
      <h3>Patient Orders</h3>
    </div>
    <table className="t-orders">
      <tbody>
        <tr>
          <th>Date</th>
          <th>Details</th>
          <th>Status</th>
          <th>Actions</th>
        </tr>
        <OrdersTable />
      </tbody>
    </table>
  </div>
);

export default AllOrders;
