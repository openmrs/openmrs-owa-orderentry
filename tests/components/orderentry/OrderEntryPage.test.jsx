import React from 'react';
import ConnectedOrderEntryPage, { OrderEntryPage } from '../../../app/js/components/orderEntry/OrderEntryPage';
import * as orderTypes from '../../../app/js/components/orderEntry/orderTypes';
import DrugOrderEntry from '../../../app/js/components/drugOrderEntry';

let props;

let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<SearchDrug {...props} />);
  }
  return mountedComponent;
};

describe('Test for Order entry page when orderentryowa.encounterType is set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: 'Admin role',
        roleError: '',
      },
      dateFormatReducer: {
        dateFormat: 'DD-MMM-YYYY HH:mm',
        error: '',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });
  it('should render component', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent).toMatchSnapshot();
  });

  it('should render Loader Images ', () => {
    props.outpatientCareSetting = null;
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div img').length).toBe(1);
  });
  it('should not show error', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(0);
  });
  it('should switch the order type in the state', () => {
    const component =  shallow(<OrderEntryPage {...props} />);
    const componentInstance = component.instance();
    const orderTypesAsArray = Object.values(orderTypes);
    expect(component.state('currentOrderType')).toBe(orderTypesAsArray[0]);
    componentInstance.switchOrderType(orderTypesAsArray[1]);
    expect(component.state('currentOrderType')).toBe(orderTypesAsArray[1]);
  });
});

describe('Test for Order entry page when orderentryowa.encounterType is not set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: 'Admin role',
        roleError: '',
      },
      dateFormatReducer: {
        dateFormat: 'DD-MMM-YYYY HH:mm',
        error: '',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });
  it('should not render DrugOrderEntry Component ', () => {
    props.settingEncounterTypeReducer = {
      settingEncounterType: '',
      error: 'Property can not be found',
    }
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find(DrugOrderEntry).length).toBe(0);
  });
  it('should show error', () => {
    props.settingEncounterTypeReducer = {
      settingEncounterType: '',
      error: 'Property can not be found',
    }
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(1);
  });
});

describe('Test for Order entry page when orderentryowa.encounterRole is set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: 'Admin role',
        roleError: '',
      },
      dateFormatReducer: {
        dateFormat: 'DD-MMM-YYYY HH:mm',
        error: '',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });

  it('should not show error', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(0);
  });
});

describe('Test for Order entry page when orderentryowa.encounterRole is not set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: '',
        roleError: 'error error',
      },
      dateFormatReducer: {
        dateFormat: 'DD-MMM-YYYY HH:mm',
        error: '',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });

  it('should not render DrugOrderEntry Component ', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find(DrugOrderEntry).length).toBe(0);
  });
  it('should show error', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(1);
  });
});

describe('Test for Order entry page when orderentryowa.dateAndTimeFormat is set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: 'Admin role',
        roleError: '',
      },
      dateFormatReducer: {
        dateFormat: 'DD-MMM-YYYY HH:mm',
        error: '',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });
  it('should not show error', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(0);
  });
});

describe('Test for Order entry page when orderentryowa.encounterRole is not set', () => {
  beforeEach(() => {
    props = {
      fetchPatientCareSetting: jest.fn(),
      getSettingEncounterType: jest.fn(),
      getSettingEncounterRole: jest.fn(),
      getDateFormat: jest.fn(),
      settingEncounterTypeReducer: {
        settingEncounterType: 'order type',
        error: '',
      },
      settingEncounterRoleReducer: {
        settingEncounterRole: 'Admin role',
        roleError: '',
      },
      dateFormatReducer: {
        dateFormat: '',
        error: 'incomplete config',
      },
      outpatientCareSetting: { uuid: '5677666' },
      inpatientCareSetting: { uuid: '6766667' },
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    };
    mountedComponent = undefined;
  });

  it('should not render DrugOrderEntry Component ', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find(DrugOrderEntry).length).toBe(0);
  });
  it('should show error', () => {
    mountedComponent = shallow(<OrderEntryPage {...props} />);
    expect(mountedComponent.find('div.error-notice').length).toBe(1);
  });
});


describe('Connected OrderEntryPage component', () => {
  it('component successfully rendered', () => {
    const store = mockStore({
      careSettingReducer: {
        outpatientCareSetting: { uuid: '' },
        inpatientCareSetting: { uuid: '' }
      },
      settingEncounterTypeReducer: {
        settingEncounterType: 'order entry',
        error: ''
      },
    });
    const props = {
      location: {search: '?patient=esere_shbfidfb_343ffd'}
    }
    const wrapper = shallow(<ConnectedOrderEntryPage store={store} {...props}/>);
    expect(wrapper.length).toBe(1);
  });
});
