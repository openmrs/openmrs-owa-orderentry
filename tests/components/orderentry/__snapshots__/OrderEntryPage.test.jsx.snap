// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Order entry page when orderentryowa.encounterType is set should render component 1`] = `
<div
  className="order-entry-page"
>
  <div>
    <Connect(PatientHeader) />
    <div
      className="header-nav"
    >
      <div>
        <h3
          className="orders-nav"
          onClick={[Function]}
          role="button"
        >
          <b>
            Orders List
          </b>
        </h3>
      </div>
      <SelectOrderType
        currentOrderType={Object {}}
        switchOrderType={[Function]}
      />
    </div>
    <RenderOrderType
      dateFormatReducer={
        Object {
          "dateFormat": "DD-MMM-YYYY HH:mm",
          "error": "",
        }
      }
      fetchPatientCareSetting={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      fetchPatientNote={
        [MockFunction] {
          "calls": Array [
            Array [
              "esere_shbfidfb_343ffd",
            ],
          ],
        }
      }
      fetchPatientRecord={
        [MockFunction] {
          "calls": Array [
            Array [
              "esere_shbfidfb_343ffd",
            ],
          ],
        }
      }
      getDateFormat={
        [MockFunction] {
          "calls": Array [
            Array [
              "default",
            ],
          ],
        }
      }
      getLabOrderables={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      getSettingEncounterRole={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      getSettingEncounterType={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      inpatientCareSetting={
        Object {
          "uuid": "6766667",
        }
      }
      location={
        Object {
          "search": "?patient=esere_shbfidfb_343ffd",
        }
      }
      outpatientCareSetting={
        Object {
          "uuid": "5677666",
        }
      }
      settingEncounterRoleReducer={
        Object {
          "roleError": "",
          "settingEncounterRole": "Admin role",
        }
      }
      settingEncounterTypeReducer={
        Object {
          "error": "",
          "settingEncounterType": "order type",
        }
      }
    />
  </div>
</div>
`;
